src/dev/platform/f3xx/Timerf3xx.cpp

```diff
@@ -142,7 +142,10 @@ void Timerf3xx::initTimer(TIM_TypeDef* timerPeripheral, uint32_t clockPeriod) {
 
 void Timerf3xx::startTimer(void (*irqHandler)(void* htim)) {
     TIM_TypeDef* timerPeripheral = this->halTimer->Instance;
-    stopTimer();
+    // If timer is not waiting to start, stop it
+    if (halTimer->State != HAL_TIM_STATE_READY)
```

Comment:
Always use braces even if you don't need them. This is liable to cause accidents later
```suggestion
    if (halTimer->State != HAL_TIM_STATE_READY) {
```